### PR TITLE
AP_HAL_ChibiOS: Add support for Bitcraze Crazyflie 2.1

### PR DESCRIPTION
**Pull request by Duccan**
This PR adds support for the Bitcraze Crazyflie 2.1 platform. The already supported Crazyflie 2.0 platform has been discontinued. 

The existing hwdef of crazyflie has been updated with the following changes:
- Changed IMU to Bosch BMI088
- Changed Baromenter to Bosch BMP388
- Changed IMU orientation to 180 degrees roll rotation 

Updated `AP_InertialSensor_BMI088.cpp`:
- Removed check return value of the soft-reset `write_register()` function. When using I2C this function returns an error for 90% out of the times. For me unknown why. Perhaps other developers can give some more insight regarding this? Removing this check definitely has not my preference..
- Increased the delay from 10 to 30 ms after the gyro reset according to the datasheet of the Bosch BMI088.

image

Datasheet:
- https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BMI088-DS001.pdf

More information (blog):
- https://discuss.ardupilot.org/t/crazyflie-goes-ardupilot/57401/4

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/crazyflie2/hwdef.dat [hunk 1]
```diff
@@ -99,10 +99,10 @@ define STORAGE_FLASH_PAGE 1
 FLASH_RESERVE_START_KB 64
 
 # one IMU
-IMU Invensense I2C:0:0x69 ROTATION_YAW_180
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/crazyflie2/hwdef.dat:102]: you should be able to leave both in, then we could support both old and new boards

File: libraries/AP_HAL_ChibiOS/hwdef/crazyflie2/hwdef.dat [hunk 2]
```diff
@@ -99,10 +99,10 @@ define STORAGE_FLASH_PAGE 1
...
+IMU BMI088 I2C:0:0x18 I2C:0:0x69 ROTATION_ROLL_180
 
 # one baro, attached via I2C on IMU
-BARO LPS2XH:probe_InvensenseIMU I2C:0:0x5D 0x69
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/crazyflie2/hwdef.dat:105]: leave both of these in too

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI088.cpp [hunk 3]
```diff
@@ -223,10 +223,9 @@ bool AP_InertialSensor_BMI088::gyro_init()
         return false;
     }
 
-    if (!dev_gyro->write_register(REGG_BGW_SOFTRESET, 0xB6)) {
-        return false;
-    }
-    hal.scheduler->delay(10);
+    // Soft-reset gyro
```
Comment (tridge) [inline libraries/AP_InertialSensor/AP_InertialSensor_BMI088.cpp:226]: the BMI088 is a very odd sensor, and we've found on other boards that it needs multiple tries to initialise. I'm ok with this change, but I'd like a comment explaining that this call may fail

**Comment by rmackay9** [comment]
@Duccan, if you would like someone to have a look at the tune and give advice please feel free to ping me somehow... you could just comment below and ideally include an onboard log of a slightly aggressive flight.

File: libraries/AP_HAL_ChibiOS/hwdef/crazyflie2/hwdef.dat (hunk 1 above)
Comment (Duccan) [reply libraries/AP_HAL_ChibiOS/hwdef/crazyflie2/hwdef.dat:102]: Ok. Added in latest commit

File: libraries/AP_HAL_ChibiOS/hwdef/crazyflie2/hwdef.dat (hunk 2 above)
Comment (Duccan) [reply libraries/AP_HAL_ChibiOS/hwdef/crazyflie2/hwdef.dat:105]: Ok. Added in latest commit

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI088.cpp (hunk 3 above)
Comment (Duccan) [reply libraries/AP_InertialSensor/AP_InertialSensor_BMI088.cpp]: Ok. Added in latest commit.

**Comment by Duccan** [comment]
@rmackay9, the drone flies decent now. I have added a defaults.parm with my PID settings. I would love to share a log file, but I do not have the SD card expansion board for the Crazyflie. There is no logging available. I could share a tlog file though.

**Comment by tridge** [comment]
merging despite CI failures, as CI failures are not related to the change, and we have flapping tests at the moment

**Comment by rmackay9** [comment]
This is included in Copter-4.0.4-rc2

**Pull request merged**